pkg/cri/store/stats/stats.go

```diff
@@ -16,12 +16,56 @@
 
 package stats
 
-import "time"
-
 // ContainerStats contains the information about container stats.
 type ContainerStats struct {
-	// Timestamp of when stats were collected
-	Timestamp time.Time
+	ContainerCPUStats
+	ContainerMemoryStats
+	ContainerFileSystemStats
+}
```

Comment:
All stats whose additional metrics will eventually be returned via Prometheus format. Rationale would be to add the additional caching logic in the future, and since stats and metrics will be set in the same place (reading the same linux cgroup file) we can read from the container stats store or sandbox stats store.

Reply:
Maybe my question wasn't clear.  What's the reason for embedding versus using fields?

```go
// embedding
type ContainerStats struct {
	ContainerCPUStats
	ContainerMemoryStats
	ContainerFileSystemStats
}

// fields
type ContainerStats struct {
	CPU        ContainerCPUStats
	Memory     ContainerMemoryStats
	FileSystem ContainerFileSystemStats
}
```